src/components/ds/button/button.tsx

```diff
@@ -0,0 +1,30 @@
+import styles from "./button.module.css";
+import { Loading } from "../../icons/loading";
+import classNames from "classnames";
+
+const Button: React.FC<{
+  handleClick?: () => void;
+  text: string;
+  icon?: JSX.Element;
+  loading?: boolean;
+  disabled?: boolean;
+  variant?: "affirm" | "decline" | "build" | "download";
```

Comment:
```suggestion
export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
    variant?: "primary" | "secondary" | "success" | "error";
    size: "small" | "medium" | "large";
    icon?: JSX.Element;
    loading?: boolean;
}

const Button: React.FC<ButtonProps>
```

Extending the native button props like this means we can accept all the props buttons can take by default like `onClick`,  `disabled` and `children`